tests/harness/harness.proto

```diff
@@ -14,4 +14,5 @@ message TestResult {
     bool   Error  = 2;
     string Reason = 3;
     bool   AllowFailure = 4;
+    repeated string AllReasons = 5;
```

Comment:
Once we add support for multiple errors in all languages, I think we'll want to get rid of the duplication between Reason and AllReason. What do you think of making Reason repeated ([which is wire-compatible with singular](https://stackoverflow.com/questions/45849903/can-a-proto3-optional-field-be-changed-to-repeated-without-breaking-wire-compati)) and adding a `bool CheckMultipleErrors` that is set by the Go (and later C++, Java, Python) binaries that we can drop once it is set for all languages?

Reply:
Thinking about this more, I'm now less sure this is the right approach. See PR comments.

Reply:
As to `repeated`, I didn't realize that, thanks! However, if we go that way, I don't currently see a way how the executor could detect if the language's executable does or does not support "lazy mode" based just on that. Just treating `len(Reasons)==1` as "does not support lazy mode" seems wrong, given that many cases do actually expect 1 failure. Would you prefer to add an extra field e.g. ".SupportsMultiError" in TestResult? Or do you have some other idea/preference? Hm, or would you prefer if the language's executable took the `.Failures` field from parsed TestCase, and compared the count on their side? (I assume that would lead to removing the comparison from the executor.) If I'm misunderstanding, I'd be super grateful for further guidance!

As to doing an extra check on the first error, I'll add that in the Go language's executable. I don't think I'm doing this now in the executor, so I may well be misunderstanding something in your suggestion; I think it might be easier for us to talk about this further once I submit the new commit; if it's gonna be not what you meant, please tell me, I'm perfectly ok with pulling commits back or amending them or whatever.

Reply:
Sorry for not being clear. Yeah, your `.SupportsMultiError` field on TestResult is what I was calling `CheckMultipleErrors`. We need a way for the test harness to know that for some languages, it should expect the executable to return all errors, and for others just the one. Options are

1. hard-code the expectation per-language in the harness
2. include that information in the TestResult

Hard-coding per-language mean checking the `TestCase.Failures` field and then modifying it based on the language (i.e. for C++ if `.Failures = 5`, treat it as 1). Including the information in the TestResult means we can upgrade each language's support independently without touching the harness again, and without hard-coding. As we upgrade each language we make the test executable also set the field, and rely on the default value (unset by un-upgraded language executables) being false. Once all languages support multiple errors, we can remove the field entirely and get rid of the false path.

Reply:
I guess I'll have to modify the other languages anyway, even if the modification is wire-compatible, given that the code for writing a `TestResult` in tests is auto-generated in CI anyway for all languages, so the modification of the `.Result[s]` field in `.proto` will spill and break the tests, I presume

Reply:
Good point, I totally forgot about that. Thanks!